src/WAV.jl

```diff
@@ -517,7 +519,7 @@ function read_data(io::IO, chunk_size, fmt::WAVFormat, format, subrange)
     # "format" is the format of values, while "fmt" is the WAV file level format
     convert_to_double = x -> convert(Array{Float64}, x)
 
-    if subrange === Void
+    if subrange === nothing || subrange === Nothing # Nothing kept for legacy reasons only
```

Comment:
I've changed the default value from the type `Void` (no `Nothing`) to its singleton value `nothing`, which is much more idiomatic. In case anyone explicitly used `subrange=Void`, this will be handled here. Should there be a `depwarn` instead of silently accepting?

Reply:
I think that you can break the API here. There is another breaking change on master that will be released with Julia 1.0.

Reply:
> I think that you can break the API here.

Ok. With a deprecation or hard-breaking? Also, on second thought, `subrange=:` might be even better than `subrange=nothing`. Preferences?

> There is another breaking change on master that will be released with Julia 1.0.

But you do plan on releasing a version that works with Julia 0.7 before 1.0 comes out, right?

Reply:
Yeah, I do plan to make a release that supports both. Do the Julia deve still plan to make 0.7 api compatible with 1.0? I haven’t been following as closely as I once did.

Reply:
Yes, 1.0 should only drop functionality that's already deprecated in 0.7. So with this PR, WAV.jl should be ready for Julia 1.0 (fingers crossed).

Meanwhile, what's your opinion on `subrange=:` vs. `subrange=nothing` and deprecation yes/no?

Reply:
I prefer `subrange=:`. You can add a deprecation if it isn't too much of a hassle. I would keep the deprecation for the Julia 0.7 release cycle and remove it when Julia 1.0 ships.